stl/inc/xpolymorphic_allocator.h

```diff
@@ -144,7 +144,8 @@ namespace pmr {
         _NODISCARD __declspec(allocator) void* allocate(_CRT_GUARDOVERFLOW const size_t _Bytes,
             const size_t _Align = alignof(max_align_t)) { // allocate _Bytes bytes of memory with alignment _Align
             _STL_ASSERT(_Is_pow_2(_Align), "memory_resource::allocate(): Alignment must be a power of two.");
-            return do_allocate(_Bytes, _Align);
+            void* _Ptr = do_allocate(_Bytes, _Align);
+            return launder(new (_Ptr) byte[_Bytes]);

```

Comment:
I don't think using `std::launder` is right. As per [[ptr.launder]/3](https://eel.is/c++draft/ptr.launder#3), the result only points to a `std::byte` object at the beginning of the storage, not "a suitable created object in the allocated region of storage". Note that the "suitable created object" is of an unspecified type which depends on following usage (as per [[intro.object]/11](https://eel.is/c++draft/intro.object#11)).
IMO the resolution is not portably implementable. If there were a (currently disallowed) hypothetical `std::launder<void*>` whose result points to such an object of an unspecified type, then it might be used for the resolution.

If we remove `launder` call here, some valid usages of the result of `allocate` would technically have UB, after the changes of object model in WG21-P0137.
IMO such UB is introduced to allow some object-structure-based alias analysis depicted in WG21-N4303. And currently MSVC and clang haven't enable such aggresive optimizations:
- on MSVC, `launder` has no effect (which means that pointer values behave as if they are `launder`'d);
- on clang, `launder` only behaves as a barrier for devirtualization (because placement `new` can replace vtables).

So currently it might be still OK to live with this theoretical UB.

I think we should drop `launder` at this moment. If `launder` gets extended in future, it'll be suitable to add it back.